jerry-core/api/jerry.c

```diff
@@ -5740,7 +5742,7 @@ jerry_arraybuffer_write (const jerry_value_t value, /**< target ArrayBuffer or S
   jerry_assert_api_available ();
 
 #if JERRY_BUILTIN_TYPEDARRAY
-  if (!(ecma_is_arraybuffer (value) || ecma_is_shared_arraybuffer (value)))
```

Comment:
Was wondering if maybe we could change the implementation of the `ecma_is_shared_arraybuffer` method to return false if the SharedArrayBuffer support is not enabled. That way we could keep most of these methods as it is (no need for the extra ecma_is_array_or_... method). Still I'm not sure that's the best approach at the moment. What do others think? @zherczeg @rerobika ?

Reply:
I am neutral, I like both actually.